modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/EditItem.js

```diff
@@ -143,45 +154,29 @@ function EditItem() {
 
 export default EditItem;
 
-const Options = ({items, selectedItem}) => {
-	const [selectedOptions, setSelectedOptions] = useState(() =>
-		JSON.parse(selectedItem.options).reduce(
-			(acc, {key, value}) => ({...acc, [key]: value}),
-			{}
-		)
-	);
-
-	return items.map(({id, key, name, productOptionValues}) => (
-		<ClayForm.Group key={id}>
-			<label htmlFor={`${id}-${name}`}>
-				{name}
-
-				<span className="reference-mark">
-					<ClayIcon symbol="asterisk" />
-				</span>
-			</label>
-
-			<ClaySelect
-				id={`${id}-${name}`}
-				onChange={({target: {value}}) =>
-					setSelectedOptions((selectedOptions) => ({
-						...selectedOptions,
-						[key]: value,
-					}))
-				}
-			>
-				{productOptionValues.map((item) => (
-					<ClaySelect.Option
-						key={item.id}
-						label={item.name}
-						selected={selectedOptions[key] === item.key}
-						value={item.key}
-					/>
-				))}
-			</ClaySelect>
-		</ClayForm.Group>
-	));
-};
+const Options = ({channelId, namespace, options, productId, selectedItem}) =>
+	options.map((option) => {
+		let Component = ProductOptionSelect;
```

Comment:
There are other fieldTypes, can we handle all of them?

Reply:
In this way other fieldTypes will be rendered as a select. I also fear that some of them won't work at all and break the page